### PR TITLE
Remove unused throws

### PR DESCRIPTION
**Pull request by davidxia**

### CODE REVIEW

**Comment by rculbertson** [comment]
:+1:

**Comment by davidxia** [comment]
replaced by https://github.com/spotify/helios/pull/582

File: helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java [hunk 1]
```diff
@@ -1597,8 +1596,7 @@ public Deployment undeployJob(final String host, final JobId jobId, final String
                                                        final String host,
                                                        final Deployment deployment,
                                                        final String token)
-      throws JobDoesNotExistException, JobAlreadyDeployedException, HostNotFoundException,
-             JobPortAllocationConflictException, TokenVerificationException {
+      throws JobDoesNotExistException, JobAlreadyDeployedException, TokenVerificationException {
```
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java]: We aren't checking if a host exists here. We check in `getUndeployOperations()`. Should we check again here or wait for `rollingUpdateAwaitRunning()` to time out?

The chances of an agent going away between undeploy and await running are slim. But if it does happen, the error message in the deployment group status will be clearer than just `timed out while retrieving job status`.

**Comment by rohansingh** [comment]
Confused. Is this PR supposed to be open or is it obsoleted?

**Comment by davidxia** [comment]
It's open. Yea I changed the status a bunch.

**Pull request merged**